coral-component-table/src/scripts/Table.js

```diff
@@ -1218,6 +1218,17 @@ class Table extends BaseComponent(HTMLTableElement) {
     const handle = this.querySelector('[coral-table-select]');
     
     if (handle) {
+
+      // If the handle is a checkbox but lacks a label, label it with "Select All".
+      if (handle instanceof Checkbox) {
```

Comment:
I'd rather use DOM attributes in case the Checkbox is not defined yet. So e.g. check for `tagName === 'coral-checkbox'` and use e.g `handle.setAttribute('labelled', i18n.get('Select All'))` instead of the property.